### PR TITLE
Breaking change in SelectField component: simplification & MultipleSelectField split

### PR DESCRIPTION
**Pull request by vmilan**
# Description

Shortcut: https://app.shortcut.com/cartoteam/story/303604/core-refactor-selectfield
[sc-303604]

Split functionality into two different components, one for multiple select, more restringed based on design specs.
The other one is a more generic select field with the addition of a placeholder.

### CODE REVIEW

File: packages/react-ui/src/components/atoms/SelectField.js [hunk 1]
```diff
@@ -1,77 +1,61 @@
...
           multiple: multiple,
-          displayEmpty: !!placeholder,
+          displayEmpty: displayEmpty || !!placeholder,
           size: size,
-          renderValue: (selected) => {
-            if (selected.length === 0) {
-              return (
-                <BoxContent ml={multiple && paddingSize}>
+          renderValue:
+            customRenderValue ||
```
Comment (mdeividas) [inline packages/react-ui/src/components/atoms/SelectField.js:41]: maybe the same here? Let's move it into `useCallback`?

**Review by mdeividas — APPROVED** [review]
LGTM

File: UPGRADE.md [hunk 2]
```diff
@@ -213,13 +213,22 @@ For external use: `import { PasswordField } from '@carto/react-ui';`.
 
 ### SelectField
 
-This component add the `placeholder` and `multiple selection` logic on top of TextField Mui component.
+This component add the `placeholder` logic on top of TextField Mui component.
```
Comment (VictorVelarde) [inline UPGRADE.md:216]:
```suggestion
This component adds the `placeholder` logic on top of TextField Mui component.
```

**Comment by VictorVelarde** [comment]
Overall LGTM, but let's have the PR in cloud-native already adapted before landing

**Pull request merged**